File: android/src/main/kotlin/io/getimpulse/player/plugin/core/PluginConstants.kt
package io.getimpulse.player.plugin.core

@Suppress("ConstPropertyName")
internal object PluginConstants {

    const val PluginTag = "impulse-player-plugin"
    const val ViewTag = "impulse-player-view"

    object Method {
        const val Load = "load"
        const val Play = "play"
        const val Pause = "pause"
        const val Seek = "seek"
        const val GetState = "get_state"
        const val IsPlaying = "is_playing"
        const val GetProgress = "get_progress"
        const val GetDuration = "get_duration"
        const val GetError = "get_error"

        const val SetAppearance = "set_appearance"
        const val SetSettings = "set_settings"
    }

    object Parameter {
        const val PictureInPictureEnabled = "picture_in_picture_enabled"
        const val CastReceiverApplicationId = "cast_receiver_application_id"
        const val Id = "id"
        const val Url = "url"
        const val Headers = "headers"
        const val Title = "title"
        const val Subtitle = "subtitle"
        const val Time = "time"
        const val Message = "message"
    }

    object Value {
        const val Normal = "normal"
        const val Italic = "italic"
    }

    object Callback {
        const val OnReady = "on_ready"
        const val OnPlay = "on_play"
        const val OnPause = "on_pause"
        const val OnFinish = "on_ready"
        const val OnError = "on_error"
    }

    object State {
        const val Loading = "loading"
        const val Ready = "ready"
        const val Error = "error"
    }
}

File: ios/Classes/Core/PluginConstants.swift
internal enum PluginConstants {
    static let pluginTag = "impulse-player-plugin"
    static let viewTag = "impulse-player-view"

    enum Method {
        static let load = "load"
        static let play = "play"
        static let pause = "pause"
        static let seek = "seek"
        static let getState = "get_state"
        static let isPlaying = "is_playing"
        static let getProgress = "get_progress"
        static let getDuration = "get_duration"
        static let getError = "get_error"
        
        static let setAppearance = "set_appearance"
        static let setSettings = "set_settings"
    }
    
    enum Parameter {
        static let pictureInPictureEnabled = "picture_in_picture_enabled"
        static let castReceiverApplicationId = "cast_receiver_application_id"
        static let id = "id"
        static let url = "url"
        static let title = "title"
        static let subtitle = "subtitle"
        static let headers = "headers"
        static let time = "time"
        static let message = "message"
    }
    
    enum Value {
        static let normal = "normal"
        static let italic = "italic"
    }

    enum Callback {
        static let onReady = "on_ready"
        static let onPlay = "on_play"
        static let onPause = "on_pause"
        static let onFinish = "on_finish"
        static let onError = "on_error"
    }

    enum State {
        static let loading = "loading"
        static let ready = "ready"
        static let error = "error"
    }
}
